msteams-platform/messaging-extensions/api-based-secret-service-auth.md

```diff
@@ -0,0 +1,137 @@
+
+1. Go to **Tools** > **API key registration**.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-api-key-registration.png" alt-text="Screenshot shows the API key registration option in Developer Portal for Teams.":::
+
+1. Select **+ New API key**.
+
+1. In the **API key registration** page, select **+ Add Secret**. A **Add an API key** dialog appears.
+
+1. Enter a value for the secret and select **Save**.
+
+   > [!NOTE]
+   >
+   > * You can maintain up to two secrets. If you need to replace one, you can do so without service interruption, as Teams will use the other configured key during the update process.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-api-key-secret.png" alt-text="Screenshot shows the Enter the value for this secret option to add a secret to the API key.":::
+
+1. Under **API key name**, add a meaningful name for the API Key. For example, API key for Contoso message extension.
+
+1. Under **Base URL**, specify a path that initiates all the API endpoints. The path must start with https, include a fully qualified domain name, and optionally, a path. Teams only transmits the secrets to endpoints where the URL begins with this value. For example, `https://api.yelp.com`. *[Mandatory]*
+
+   Base URL ensures that the key remains secure and isn't leaked to random endpoints, even if another app illicitly acquires the API secret registration ID and incorporates it into their own app. We enforce this URL constraint on API keys. If the path registered here doesn't prefix the target endpoint defined in the app manifest, the call gets dropped.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-register-key-domain.png" alt-text="Screenshot shows the Description and Add domain options in the API key registration page in Developer Portal for Teams.":::
+
+1. Under **Target tenant**, select any of the following:
+
+   * **Home tenant**: The API key is only functional within the tenant where it's registered.
+   * **Any tenant**: The API key t is usable in any tenant.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-api-key-tenant.png" alt-text="Screenshot shows the Home tenant and Any tenant options under set a target tenant heading in Developer Portal for Teams.":::
+
+1. Under **Target Teams app**, select any of the following:
+
+   * **Existing Teams app**: The **Existing Teams app** option binds the API secret registration to your specific Teams app.
+   * **Any Teams app**: The API key can be used with any Teams app.
+
+   Adding a domain ensures that the key isn't exposed to endpoints. However, the API secret registration ID is publicly accessible and can be added to random apps, potentially sending unwanted data to a developer's endpoint. To prevent this, you can bind the registration to a specific app and Teams rejects requests for any app other than the one specified in the secret registration.
```

Comment:
Adding a domain ensures that the key isn't exposed to random endpoints

Reply:
Updated as suggested